### PR TITLE
Improve generate report

### LINKED ISSUE #2422: Update the introduction text of the reports
When generating a report we currently have the following global introduction text across all reports. 

```
Through a systematic examination of the network's infrastructure, the scan has identified potential security weaknesses and points of exposure. Additionally, the report offers actionable recommendations and prioritizes remediation steps to fortify the network's defenses against potential threats. This invaluable assessment serves as a roadmap for enhancing network security, ensuring the confidentiality, integrity, and availability of critical data and resources. 
```

As this text is the same in all reports and currently not accurate to the contents of what we we show, this introduction text should be updated.

### LINKED ISSUE #2588: Update HTML of 'Generate Report' conform design
### Problem
Currently the Generate Reports don't look the same as the designs. 

As seen in the screenshots below, we're currently only showing chapter 3 of the design in the reports. These parts are still missing or need to be changed:
* Update report header
* Add "Summary" section
* Add "Recommendations" section
* Add "Asset overview" section
* Add side menu

#### Design
For example, look at the Findings Report **design**:
Image

#### Actual report
And the **actual** Findings Report:
Image

### Solutions
We should update the current files and add some more, so the current reports meet the created designs.

### To do:
- [ ] Update "generate_report.html" and add a side menu
- [ ] Update "partials/report_header.html"
- [ ] Update "partials/report_introduction.html"
- [ ] Update "summary/report_summary.html"
- [ ] Add "partials/report_recommendations.html"
- [ ] Add "partials/report_asset_overview" (this will be the old "report_summary.html")

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes
This PR makes a lot of changes to (mostly) the HTML of the Generate Reports. 

The following changes are made to meet the [design ](https://www.figma.com/file/QRL8085Z827bNdO0bIDdFL/KAT-%7C-For-review?type=design&node-id=3287-21907&mode=design):
* Add a table of contents
* Update the header
* Add chapter numbers
* Update the download report button
* Change "Summary" to "Asset overview"

Other changes that have been made to the Generate Report:
* Add the human readable OOI to the titles of the reports (instead of the full OOI name)
* Add text to the report when no data has been found (insead of showing nothing/an empty report)
* Add "-" to some tables when no data has been found (instead of showing an empty field)
* Fix some headings that were too big
* Add "nowrap" class to TTL column in DNS report
* Add introductory text at the beginning of each section
* Change the order of the reports
* Fix styling of Risk Levels (severity_level)

### Issue link
Closes #2588 
Closes #2422 

#### Before
afbeelding

#### After
afbeelding
---

### Opens
This PR opens some new issues:
#2671 and #2655

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/report_types/ipv6_report/report.html [hunk 1]
```diff
@@ -25,13 +26,17 @@
...
-                                    {% endif %}
-                                {% endfor %}
+                                {% if info.systems %}
+                                    {% for system in info.systems %}
+                                        {% if not forloop.last %}
+                                            {{ system }},
+                                        {% else %}
+                                            {{ system }}
+                                        {% endif %}
+                                    {% endfor %}
```
Comment (ammar92) [inline rocky/reports/report_types/ipv6_report/report.html:36]: See `join`

File: rocky/reports/report_types/systems_report/report.html [hunk 2]
```diff
@@ -1,39 +1,63 @@
...
+                    <tr>
+                        <td>{{ ip.tokenized.address }}</td>
+                        <td>
+                            {% if ip_services.hostnames %}
+                                {% for hostname in ip_services.hostnames %}
+                                    {% if not forloop.last %}
+                                        {{ hostname.tokenized.name }},
+                                    {% else %}
+                                        {{ hostname.tokenized.name }}
+                                    {% endif %}
```
Comment (ammar92) [inline rocky/reports/report_types/systems_report/report.html:37]: See `join`

File: rocky/reports/report_types/rpki_report/report.html [hunk 3]
```diff
@@ -11,6 +13,11 @@ <h2>{% translate "Resource Public Key Infrastructure" %}</h2>
             announcements, ensuring reliable server access and secure internet traffic.
         {% endblocktranslate %}
     </p>
+    <p>
+        {% blocktranslate trimmed %}
+            The RPKI report shows if an RPKI route announcement was available for the system and if this announcement is not expired.
+        {% endblocktranslate %}
```
Comment (ammar92) [inline rocky/reports/report_types/rpki_report/report.html:19]: `translate`

**Review by ammar92 — COMMENTED** [review]
Looks good in general; just a few tips to improve it even more. There are probably more use cases of joins and blocktranslate vs translate calls that I haven't seen

**Review by ammar92 — DISMISSED** [review]
Looks good to me 👍

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Looks good! I did find some things we could improve to make it better. I've added them to the last section. I expect some/maybe all things can be picked up in different tickets. 

### What doesn't work:
n/a

### Bug or feature?:
- Takes a long time to generate the report for all 12 report types.

- Asset Overview should be sorted based on type I think where Hostnames should appear before IP addresses.
    - The object primary key "IPService|internet|<IP>|tcp|443|https" should drop the 'IPService|internet' part and only show IP address, port, service, protocol. Maybe this should be done through additional columns. 
- Selected report types should be sorted based on either the order of the reports or otherwise alphabetically (a-z). There are arguments for both. I think a-z is the easiest for implementation for now. 

- Change text from "No ciphers found for this IPService." to "No ciphers were found for this combination of IP address, port and service."

- The order of all the reports is now based on IP/hostname, I think it become more organised to sort these based on report type and within that report show the output per OOI. Unsure if this is even possible in the current setup. 


- The primary key now is something like: "TLS Report for IPService|internet|<ip>|tcp|443|https" it becomes more readable if this is something like: TLS Report for <IP> on port <port> for <service>. 
    - Primary key in Safe Cnonections report for the section name of the compliance issues table is now "IPAddressV4|internet|<ip>" It becomes more readable if this just says: IP

    - Primary keys of sections: Is it possible to split the IP (and protocol) from the URL/Hostname? Thus instead of 'https://1.2.3.4:443/tcp' we show 'https://1.2.3.4' at 'port: 443/tcp'

**Comment by madelondohmen** [comment]
>     * Change text from "No ciphers found for this IPService." to "No ciphers were found for this combination of IP address, port and service."
> 
>     * The order of all the reports is now based on IP/hostname, I think it become more organised to sort these based on report type and within that report show the output per OOI. Unsure if this is even possible in the current setup.
> 

These two should now be resolved. I have created a new ticket for the other improvements and I will create another PR for those.

File: rocky/reports/report_types/dns_report/report.html [hunk 4]
```diff
@@ -39,17 +41,23 @@ <h2>{% translate "Records found" %}</h2>
...
+                        <td class="nowrap">{{ ooi.ttl }} {% translate "minutes" %}</td>
                         <td>{{ ooi.content }}</td>
                     </tr>
                 {% endfor %}
             </tbody>
         </table>
     </div>
 {% else %}
-    <p>{% translate "No records have been found." %}</p>
+    <p>{% translate "No records have been found  on this system." %}</p>
```
Comment (underdarknl) [inline rocky/reports/report_types/dns_report/report.html:52]: regression as this was changed recently

**Review by underdarknl — APPROVED** [review]

**Pull request merged**